### PR TITLE
Move WinUI 3 to Preview 2

### PR DESCRIPTION
**Pull request by asklar**
Moving nuget packages to preview 2 and adding a couple of properties that the VSIXs added in preview 2.

Turns out my xaml load guard caught a bug :)  we were using `Windows::UI::Text::FontWeights` instead of `Microsoft::UI::Text::FontWeights` which resulted in loading WUX.dll just to get a struct, so I fixed that.

image

### CODE REVIEW

File: packages/playground/windows/playground/Playground.vcxproj [hunk 1]
```diff
@@ -79,7 +79,7 @@
   <ImportGroup Label="ReactNativeWindowsPropertySheets">
     <Import Project="$(ReactNativeWindowsDir)\PropertySheets\External\Microsoft.ReactNative.Uwp.CppApp.props"/>
   </ImportGroup>
-  <Import Project="..\packages\$(WinUIPackageProps)" Condition="'$(WinUIPackageProps)'!=''" />
+  <Import Project="..\packages\$(WinUIPackageProps)" Condition="'$(WinUIPackageProps)'!='' And Exists('$(WinUIPackageProps)')" />
```
Comment (dannyvv) [inline packages/playground/windows/playground/Playground.vcxproj:82]: Shouldn't this be prefixed with `..\packages`
```suggestion
  <Import Project="..\packages\$(WinUIPackageProps)" Condition="'$(WinUIPackageProps)'!='' And Exists('..\packages\$(WinUIPackageProps)')" />
```

File: vnext/Microsoft.ReactNative.Cxx/UI.Text.h [hunk 2]
```diff
@@ -3,8 +3,11 @@
 
 #pragma once
 
+#include <winrt/Windows.UI.Text.h>
+
 #ifdef USE_WINUI3
 #include <winrt/Microsoft.UI.Text.h>
```
Comment (dannyvv) [inline vnext/Microsoft.ReactNative.Cxx/UI.Text.h:9]: Crazy thought:
Should WinUI3 produce an entire compatibility alias-layer (perhaps both directions) for folks that need to support both?

Comment (asklar) [reply vnext/Microsoft.ReactNative.Cxx/UI.Text.h:9]: That's a noble idea :) some types/APIs don't exist in one or the other so it's not a clean rename, unfortunately. That's why you'll see blocks of code in our files with #ifdef USE_WINUI3 or #ifndef USE_WINUI3. There's more work to do to support what will be missing features in winui3 at launch (e.g. core window keyboard occlusion events, etc.)

Comment (asklar) [reply vnext/Microsoft.ReactNative.Cxx/UI.Text.h:9]: @dannyvv there is however an intern working on a conversion tool. I think the expectation is that most people just migrate once and they're done and don't have to keep supporting both, we're the odd duck here

**Pull request merged**